samples_checker/samples_checker/check_samples.py

```diff
@@ -7,43 +7,57 @@
-sample_xml = args.samplexml
-file_xml = args.filexml
-file_path = args.filepath
-
-submission_samples = utils.get_samples_from_xml(sample_xml)
-submitted_files = utils.get_files_from_xml(file_xml)
-
-has_difference = False
-for file_name in submitted_files:
-    file_type = submitted_files.get(file_name, '')
-    difference_submission_xls_submitted_file, difference_submitted_file_submission_xls =\
-        utils.get_sample_difference(submission_samples, file_path + '/' + file_name, file_type)
-
-    if difference_submitted_file_submission_xls:
-        has_difference = True
-        if file_type == 'vcf':
-            print('The submission does not contain the following genotype ids:', file=sys.stderr)
-        else:
-            print('The submission does not contain the following samples:', file=sys.stderr)
-        print(difference_submitted_file_submission_xls, file=sys.stderr)
-
-    if difference_submission_xls_submitted_file:
-        has_difference = True
-        print('The following samples are not found in file '+file_name, file=sys.stderr)
-        print(difference_submission_xls_submitted_file, file=sys.stderr)
-
-if has_difference:
-    quit(1)
-
-print('Samples checking completed!', file=sys.stdout)
+
+def get_sample_diff(file_path, file_xml, sample_xml, submission_type="t2d"):
+    submission_samples = utils.get_samples_from_xml(sample_xml)
+    submitted_files = utils.get_files_from_xml(file_xml)
+    has_difference = False
+    for file_name in submitted_files:
+        file_type = submitted_files.get(file_name, '')
+        difference_submission_xls_submitted_file, difference_submitted_file_submission_xls = \
+            utils.get_sample_difference(submission_samples, file_path + '/' + file_name, file_type)
+
+        if difference_submitted_file_submission_xls:
+            has_difference = True
+            if file_type == 'vcf':
+                if submission_type == "t2d":
+                    print('The submission does not contain the following genotype ids:', file=sys.stderr)
+                if submission_type == "eva":
+                    print('Samples that appear in the VCF but not in the Metadata sheet:', file=sys.stderr)
+            else:
+                print('The submission does not contain the following samples:', file=sys.stderr)
+            print(difference_submitted_file_submission_xls, file=sys.stderr)
+
+        if difference_submission_xls_submitted_file:
+            has_difference = True
+            if submission_type == "t2d":
+                print('The following samples are not found in file ' + file_name, file=sys.stderr)
+            if submission_type == "eva":
+                print('Samples that appear in the Metadata sheet but not in the VCF: ' + file_name,
+                      file=sys.stderr)
+            print(difference_submission_xls_submitted_file, file=sys.stderr)
+    if has_difference:
+        quit(1)
+    print('Samples checking completed!', file=sys.stdout)
```

Comment:
This message could be printed before exiting, regardless of what the exit code is.

Reply:
+1